File: solution344.cpp
class Solution {
public:
    void reverseString(vector<char>& s) {
        int n = s.size();
        int p0 = 0, p1 = n-1;
        while(p0 <= p1) {
            swap(s[p0], s[p1]);
            p0 ++;
            p1 --;
        }
    }
};

File: solution344.rs
impl Solution {
    pub fn reverse_string(s: &mut Vec<char>) {
        let (mut p0, mut p1) = (0, s.len() as i32 - 1);
        while p0 < p1 {
            s.swap(p0 as usize, p1 as usize);
            p0 += 1;
            p1 -= 1;
        }
    }
}